controller.py

```diff
@@ -14,28 +18,27 @@
 class Controller:
 
     # requestQueue is shared queue among all processes
-    #@logged_method
-    def __init__(self):
+    def __init__(self, database: DB):
+        if not isinstance(database, DB):
+            raise TypeError('Controller was not supplied a DB object')
         self.manifest = Manifest()
         self.request_queue = multiprocessing.Queue()
         self.listener = Listener(self.request_queue)
+        self.database = database
 
-    #@logged_method
-    def create_request_processor(self):
-        req = ProcessRequest(self.request_queue)
+    def _create_request_processor(self):
+        req = RequestProcessor(self.request_queue, self.database)
         req.process_requests()
 
-    #@logged_method
     def create_request_processors(self):
         processes = []
         for i in range(self.manifest.number_of_request_processors):
             logger.info(f"creating request processor {str(i)}")
             # print('Creating processes %d' % i)
-            processes.append(Process(target=self.create_request_processor))
+            processes.append(Process(target=self._create_request_processor))
```

Comment:
made `create_request_processor` private just to make it clear that it is only intended to be accessed internally.